kubernetes/falco.go

```diff
@@ -0,0 +1,473 @@
+/*
+Copyright © 2019 Kris Nova <kris@nivenly.com>
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubernetesfalc
+
+import (
+	"fmt"
+
+	"github.com/kris-nova/logger"
+	"k8s.io/api/apps/v1beta2"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/api/rbac/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+)
+
+// FalcoInstaller is a data structure used to install Falco in Kubernetes
+type FalcoInstaller struct {
+	k8s *kubernetesConfigClient
+
+	// NamespaceName is an example of a "Global" variable we certainly should have
+	NamespaceName string
+
+	// DaemonSetName is an example of a "Possible" variable we don't know if we wan't to expose
+	DameonSetName string
+}
+
+// Delete is brute force method used to permanently destroy Falco resources in Kubernetes.
+func (i *FalcoInstaller) Delete(k8s *kubernetesConfigClient) error {
```

Comment:
Right now we leave various resources installed (for instance the AuditSink) we probably want to delete every resource in backwards order